flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/rules/logical/SplitAggregateRule.scala

```diff
@@ -138,11 +139,14 @@ class SplitAggregateRule extends RelOptRule(
     val windowProps = fmq.getRelWindowProperties(agg.getInput)
     val isWindowAgg = WindowUtil.groupingContainsWindowStartEnd(agg.getGroupSet, windowProps)
     val isProctimeWindowAgg = isWindowAgg && !windowProps.isRowtime
+    val isSessionWindowAgg = isWindowAgg &&
+      windowProps.getWindowSpec.isInstanceOf[SessionWindowSpec]
     // TableAggregate is not supported. see also FLINK-21923.
     val isTableAgg = AggregateUtil.isTableAggregate(agg.getAggCallList)
 
     agg.partialFinalType == PartialFinalType.NONE && agg.containsDistinctCall() &&
-      splitDistinctAggEnabled && isAllAggSplittable && !isProctimeWindowAgg && !isTableAgg
+      splitDistinctAggEnabled && isAllAggSplittable && !isProctimeWindowAgg &&
+      !isSessionWindowAgg && !isTableAgg
```

Comment:
The logic seems to be quite messy here, and there is no explanation for the newly added session window.

Reply:
add comment to explain for why disable distinct split for session window.